### PR TITLE
reoganisation with uuid integration to manage the cases

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: AbdelHedhili <abdelsalem.hedhili@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*
Cases were identified by name

**What is the new behavior (if this is a feature change)?**
Now by uuid

### CODE REVIEW

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 1]
```diff
@@ -70,24 +68,67 @@ String getFormat(Path caseFile) {
...
+        if (caseFile != null) {
+            return caseFile;
+        }
+        throw CaseException.createDirectoryNotFound(caseUuid);
+    }
+
+    Path getCaseDirectory(UUID caseUuid) {
+        Path publicCaseDirectory = getPublicStorageDir();
+        Path privateCaseDirectory = getPrivateStorageDir();
+        Path caseFile = publicCaseDirectory.resolve(caseUuid.toString());
```
Comment (jonenst) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:97]: rename to caseDirectory ?

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 2]
```diff
@@ -70,24 +68,67 @@ String getFormat(Path caseFile) {
...
+    }
+
+    public Path walkCaseDirectory(Path caseDirectory) {
+        if (Files.exists(caseDirectory) && Files.isDirectory(caseDirectory)) {
+            try (Stream<Path> pathStream = Files.walk(caseDirectory)) {
+                List<Path> list = pathStream.filter(file -> !Files.isDirectory(file)).collect(Collectors.toList());
+                if (list.isEmpty()) {
+                    throw CaseException.createDirectoryEmpty(caseDirectory);
+                }
+                return list.get(0);
```
Comment (jonenst) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:115]: use Stream::findFirst ? 
but what about cases with multiple files ?

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 3]
```diff
@@ -70,24 +68,67 @@ String getFormat(Path caseFile) {
...
-    Optional<byte[]> getCaseBytes(String caseName) {
+    Optional<byte[]> getCaseBytes(UUID caseUuid) {
         checkStorageInitialization();
 
-        Path caseFile = getCaseFile(caseName);
+        Path caseFile;
+        try {
+            caseFile = getCaseFile(caseUuid);
+        } catch (CaseException e) {
+            return Optional.empty();
```
Comment (jonenst) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:130]: I think it's a bad design to use exceptions inside a single class like this. Exceptions are usefull for going through the callstack between two distant pieces of code. Inside one class, simpler control flows is better. Try to remove the exception ?

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 4]
```diff
@@ -99,22 +140,33 @@ Path getCaseFile(String caseName) {
...
         checkStorageInitialization();
-
-        Path caseFile = getCaseFile(caseName);
-        return Files.exists(caseFile) && Files.isRegularFile(caseFile);
+        Path caseFile;
+        try {
+            caseFile = getCaseFile(caseName);
+            return Files.exists(caseFile) && Files.isRegularFile(caseFile);
+        } catch (CaseException e) {
+            return false;
```
Comment (jonenst) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:150]: Same remark, try to remove the exception

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java (hunk 2 above)
Comment (jonenst) [reply case-server/src/main/java/com/powsybl/caseserver/CaseService.java:115]: I guess we will deal with multiple files in a next PR, the previous code was already handling only one file

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 5]
```diff
@@ -125,20 +177,25 @@ void importCase(MultipartFile mpf) {
...
         caseInfosPublisher.onNext(CaseInfos.getMessage(caseInfos));
+        return caseUuid;
     }
 
-    Optional<Network> loadNetwork(String caseName) {
+    Optional<Network> loadNetwork(UUID caseUuid) {
         checkStorageInitialization();
-
-        Path caseFile = getCaseFile(caseName);
+        Path caseFile;
```
Comment (jonenst) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:193]: try to remove the exception

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 6]
```diff
@@ -149,48 +206,63 @@ void importCase(MultipartFile mpf) {
...
+    void deleteCase(UUID caseUuid) {
+        checkStorageInitialization();
+        Path caseDirectory = getCaseDirectory(caseUuid);
+        deleteDirectoryRecursively(caseDirectory);
+    }
+
+    void deleteAllCases() {
+        checkStorageInitialization();
+
+        List<Path> directories = new ArrayList<>();
```
Comment (jonenst) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:233]: Arrays.asList(getPrivateStorageDir(), getPublicStorageDir())

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 7]
```diff
@@ -99,22 +136,31 @@ Path getCaseFile(String caseName) {
...
-
         Path caseFile = getCaseFile(caseName);
+        if (caseFile == null) {
+            return false;
+        }
         return Files.exists(caseFile) && Files.isRegularFile(caseFile);
     }
 
-    void importCase(MultipartFile mpf) {
+    UUID importCase(MultipartFile mpf) {
```
Comment (geofjamg) [inline case-server/src/main/java/com/powsybl/caseserver/CaseService.java:148]: Something bother me here: there is no way to import a case in the public part using the REST API? 
Regarding the ongoing work on FTP cron to feed case server with public data, how will data be sent to the case server?
Maybe we should add an option to the import to decide if we import it on public or private part.

**Pull request merged**